dklearn/wrappers.py

```diff
@@ -0,0 +1,256 @@
+                X = X.rechunk((X.chunks[0], X.shape[1]))
+            if n_classes is None or n_classes == 2:
+                res = X.map_blocks(func, chunks=(X.chunks[0],), drop_axis=1)
+            else:
+                res = X.map_blocks(func, chunks=(X.chunks[0], n_classes))
+        elif isinstance(X, db.Bag):
+            res = X.map_partitions(func)
+        elif is_list_of(X, Delayed):
+            func = delayed(func, pure=True)
+            res = [func(i) for i in X]
+            if compute:
+                return list(dask.compute(*res))
+            return res
+        else:
+            return func(X) if compute else delayed(func, pure=True)(X)
+        return res.compute() if compute else res
+
+    @if_delegate_has_method('estimator')
+    def score(self, X, y, sample_weight=None, compute=True):
+        if compute:
+            X, y, sample_weight = dask.compute(X, y, sample_weight)
+            return self.estimator.score(X, y, sample_weight=sample_weight)
+        else:
+            return score_chunk(self.estimator, X, y, sample_weight)
+
+
+@delayed(pure=True)
+def fit_chunk(est, X, y, **kwargs):
+    return clone(est).fit(X, y, **kwargs)
+
+
+@delayed(pure=True)
+def average_coefs(estimators):
+    if len(estimators) == 1:
+        return estimators[0]
+    o = clone(estimators[0])
+    is_clf = is_classifier(o)
+    if is_clf:
+        classes = np.unique(np.concatenate([m.classes_ for m in estimators]))
+        o.classes_ = classes
+        n_classes = len(classes)
+    if not is_clf or all(m.classes_.size == n_classes for m in estimators):
+        o.coef_ = np.mean([m.coef_ for m in estimators], axis=0)
+        o.intercept_ = np.mean([m.intercept_ for m in estimators], axis=0)
+    else:
+        # Not all estimators got all classes. Multiclass problems are fit using
+        # ovr, which results in a row per class. Here we average the
+        # coefficients for each class, using zero if that class wasn't fit.
+        n_features = estimators[0].coef_.shape[1]
+        coef = np.zeros((n_classes, n_features), dtype='f8')
+        intercept = np.zeros(n_classes, dtype='f8')
+        for m in estimators:
+            ind = np.in1d(classes, m.classes_)
+            coef[ind] += m.coef_
+            intercept[ind] += m.intercept_
+        o.coef_ = coef / len(estimators)
+        o.intercept_ = intercept / len(estimators)
+    return o
+
+
+class Averaged(DaskWrapper):
+    """Wrap an estimator by averaging coefficients"""
+    def _fit(self, X, y, **kwargs):
+        if not hasattr(self.estimator, 'fit'):
+            raise ValueError("estimator must support `fit`")
+        X, y = check_X_y(X, y)
+        x_parts, y_parts = as_lists_of_delayed(X, y)
+        est = self.estimator
+        chunks = [fit_chunk(est, xp, yp, **kwargs) for (xp, yp) in
+                  zip(x_parts, y_parts)]
+        return average_coefs(chunks)
+
+
+@delayed(pure=True)
+def _unique_chunk(x):
+    return np.unique(x)
```

Comment:
This would come through prettier in diagnostics like the progress.  

There is also some cleanliness value, I think, in just using numpy/pandas functions when they serialize well.